asyncpg/protocol/coreproto.pyx

```diff
@@ -315,6 +356,17 @@ cdef class CoreProtocol:
             msg = self.buffer.read_cstr()
         self.result_status_msg = msg
 
+    cdef _parse_copy_data_msgs(self):
+        cdef:
+            ReadBuffer buf = self.buffer
+
+        self.result = buf.consume_messages(b'd')
+
+        self._skip_discard = True
+        if not buf.has_message():
```

Comment:
We'll need a comment here, explaining that we either:

1. have read all messages from the buffer (it had only `d` messages) and now we need push the read data back;

2. there are unprocessed messages in the buffer, so we want to let `_process__copy_out_data` to process them and push the result.